### DISCUSSION TITLE
Can someone explain what is happening here?

### DISCUSSION THREAD

**Discussion by jsiddall** [OpenHantek/OpenHantek6022#331] [Q&A] [answered]
I am confused by some behavior I am seeing.  This is latest stable version of OpenHantek6022 in demo mode with only CH2 (the 500 Hz square wave) and its spectrum enabled.  When I set the timebase to 10 ms/div everything looks good:

![20221025_104940](https://user-images.githubusercontent.com/7401394/197807031-a36f92e7-c0f9-4c1b-bee8-9534c367d9e8.png)

Nice square wave, and a spectrum with odd harmonics at multiples of 500 Hz.  But if I increase the timebase to 20 ms/div things go wonky:

![20221025_105850](https://user-images.githubusercontent.com/7401394/197809236-72ee2e89-e613-4fb5-9677-f268e8c045d6.png)

Specifically, the spectrum now shows everything happening as if the square wave were actually at 250 Hz, including odd harmonics at multiples of 250 Hz.  The sample rate is 50k/s, which is 100x the base frequency though, being a square wave, I realize the harmonics go on forever.

At first I thought this was a bug, but it may be that I just don't understand what is happening.  Thanks for any insight!

**Comment by Ho-Ro** [comment, marked as answer]
> Can someone explain what is happening here?

I do.

At sampling rates < 100 kS/s, the frequency of the demo signal is reduced to show at least some recognisable traces instead of a blurred colour band - especially in "roll" mode. Also, at very high sampling rates > 10 MS/s, the channel 1 frequency starts to increase from 1000 Hz to 1200 and 1500 Hz, the same for the 500 Hz signal of channel 2 - if I remember correctly due to rounding during integer calculation. In any case, the frequency display at the bottom right is correct.
I originally added this demo mode as a hack to create a reproducible test signal for function development. Take it as it is.

BTW: If you wonder what this artifact could be - it is the histogram of the signal to the left.
![image](https://user-images.githubusercontent.com/12542359/197823466-935e032a-b9e3-45f1-a85d-50ad6fe813ce.png)
It can be switched off with this button (or use keyboard shortcut H)
![image](https://user-images.githubusercontent.com/12542359/197823734-616ce893-6110-4f08-a0bf-c76a141e0583.png)

Do you intend to use a real Hantek6022?

**Reply by jsiddall** [reply]
Got it, that makes sense.

Yes, I do have a real 6022 on order, just trying to familiarize myself with the software while waiting.

Thanks for the quick reply, great application and support.  Looking forward to doing some real measuring!

**Reply by Ho-Ro** [reply]
Great - if you like to do some HW improvements, just check the [AC modification](https://github.com/OpenHantek/OpenHantek6022/blob/main/docs/HANTEK6022_AC_Modification.pdf) and the [FG modification](https://github.com/OpenHantek/OpenHantek6022/blob/main/docs/HANTEK6022_Frequency_Generator_Modification.pdf).
And when you're using Ubuntu [these tools](https://github.com/Ho-Ro/Hantek6022API) are easy to install and can be helpful.
